RecoTracker/TkDetLayers/src/TBLayer.cc

```diff
@@ -51,14 +51,19 @@ SubLayerCrossings TBLayer::computeCrossings(const TrajectoryStateOnSurface& star
 
   HelixBarrelCylinderCrossing innerCrossing(
       startPos, startDir, rho, propDir, *theInnerCylinder, HelixBarrelCylinderCrossing::onlyPos);
-  if (!innerCrossing.hasSolution())
-    return SubLayerCrossings();
 
   HelixBarrelCylinderCrossing outerCrossing(
       startPos, startDir, rho, propDir, *theOuterCylinder, HelixBarrelCylinderCrossing::onlyPos);
-  if (!outerCrossing.hasSolution())
+
+  if (!innerCrossing.hasSolution() && !outerCrossing.hasSolution())
```

Comment:
the main difference by this point is the extra cost of computing the outerCrossing.
Looking at out monitoring for PU200 : track building takes up ~900 units vs about 2 units in ```TBLayer::computeCrossings```.
I think that this is a somewhat acceptable increase in cost.
Still, it would be better to add a  check that  startPos is close enough to care. 
[pseudo-coding, mind the typos]
One can use the following test to see if the starting point is in the gap between two cylinders 
```
(theOuterCylinder->position() - startPos).perp() < theOuterCylinder->radius() && (theInnerCylinder->position() - startPos).perp() > theInnerCylinder->radius() 
```
(this assumes the cylinders are aligned, perhaps it's good enough?)

Reply:
I verified I get the same exact results without this short circuit on my 100k event signal sample.
```cpp
  bool inBetween = ((theOuterCylinder->position() - startPos).perp() < theOuterCylinder->radius())
    && ((theInnerCylinder->position() - startPos).perp() > theInnerCylinder->radius());

  HelixBarrelCylinderCrossing innerCrossing(
      startPos, startDir, rho, propDir,*theInnerCylinder, HelixBarrelCylinderCrossing::onlyPos);
  if (!inBetween && !innerCrossing.hasSolution()) return SubLayerCrossings();

  HelixBarrelCylinderCrossing outerCrossing(
      startPos, startDir, rho, propDir,*theOuterCylinder, HelixBarrelCylinderCrossing::onlyPos);

  if (!innerCrossing.hasSolution() && outerCrossing.hasSolution()) {
    innerCrossing = outerCrossing;
  } else if (!outerCrossing.hasSolution() && innerCrossing.hasSolution()) {
    outerCrossing = innerCrossing;
  }
```
If this is what you had in mind, I'll update the PR

Reply:
yes, it looks like what I wanted (in the sense of having a possible/likely return before the ```outerCrossing``` is created).